pulp_service/pulp_service/app/tasks/package_scan.py

```diff
@@ -78,7 +93,10 @@ def _get_content_from_repo_version(repo_version_pk: str):
     repo_version = RepositoryVersion.objects.get(pk=repo_version_pk)
     for content_unit in repo_version.content:
         content = content_unit.cast()
-        ecosystem = _identify_package_ecosystem(content)
+        try:
+            ecosystem = _identify_package_ecosystem(content)
+        except Exception as e:
+            content_queue.put(e)
```

Comment:
Consider making a decorator that catches any exception, puts it into the content_queue.put() and the re-raises it and then decorate both `_get_content_from_repo_version` and `_parse_npm_pkg_dependencies`. I think it would be better to have the "exception propogation to MainThread" cover the entire background thread code, not just the errors you expect. Decorating "the whole thread's entrypoint" would do that in a clean, reusable way.